server/controllers/users.js

```diff
@@ -309,22 +312,39 @@ module.exports = (app) => {
           })
           .catch(next);
       } else if (req.query.profile) {
-        var groups;
-        req.user
-        .getGroups()
-        .then(results => {
-          groups = results.map(g => Object.assign(g.minimal, { role: g.UserGroup.role, createdAt: g.UserGroup.createdAt }));
-          return groups;
+        var groupInfoArray = []
+        req.user.getGroups()
+        .then(groups => {
+          return Promise.all(groups.map(group => {
+            return Promise.all([
+              group.getYearlyIncome(),
+              new Promise((resolve, reject) => {
+                groupsController.getUsers({group: group}, {send: resolve}, reject);
```

Comment:
We try not to link controllers to each other, especially by faking the req, res, next params. It might be better to move this in a library, since you only need backers (from what I can tell on the frontend).